### PR TITLE
Release Proposal: v2.12

### PR DESCRIPTION
**Pull request by avade**

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Service Broker API Release Notes #
 
+## <a id='2-12'></a>v2.12 ##
+2017-06-13
+
+* Added `context` field to request body for provisioning a service instance (`PUT /v2/service_instances/:instance_id`) and updating a service instance (`PATCH /v2/service_instances/:instance_id`). Also added a `profile.md` file describing recommended usage paterns for environment-specific extensions and variations.
+* The specification now uses RFC2119 keywords to make it clearer when things are required rather than optional or recommended.
```
Comment (pmorie) [inline release-notes.md]: Are these intended to be complete?  I think everything in this list needs to be represented here: https://github.com/openservicebrokerapi/servicebroker/milestone/1?closed=1

https://github.com/openservicebrokerapi/servicebroker/pull/178 is an especially important one

File: spec.md [hunk 2]
```diff
@@ -814,7 +815,7 @@ The request provides these query string parameters as useful hints for brokers.
 ##### cURL
 ```
 $ curl 'http://username:password@broker-url/v2/service_instances/:instance_id?accepts_incomplete=true
-  &service_id=service-id-here&plan_id=plan-id-here' -X DELETE -H "X-Broker-API-Version: 2.11"
+  &service_id=service-id-here&plan_id=plan-id-here' -X DELETE -H "X-Broker-API-Version: 2.12"
```
Comment (pmorie) [inline spec.md]: I believe there are two missing cURL sections in this version of the diff.  I count 5 and there should be 7

File: spec.md [hunk 3]
```diff
@@ -84,17 +84,18 @@ added to enable support for new features.
 These fields MUST be OPTIONAL and SHOULD be ignored by clients and servers
 that do not understand them.
 
-### Changes Since v2.10
+### Changes Since v2.11
 
-* Add `bindable` field to [Plan Object](#plan-object) to allow services to have both bindable and non-bindable plans.
+* Added `context` field to request body for provisioning a service instance (`PUT /v2/service_instances/:instance_id`) and updating a service instance (`PATCH /v2/service_instances/:instance_id`). Also added a `profile.md` file describing RECOMMENDED usage paterns for environment-specific extensions and variations.
```
Comment (pmorie) [inline spec.md]: typo: patterns

File: spec.md [hunk 4]
```diff
@@ -84,17 +84,18 @@ added to enable support for new features.
 These fields MUST be OPTIONAL and SHOULD be ignored by clients and servers
 that do not understand them.
 
-### Changes Since v2.10
```
Comment (pmorie) [inline spec.md]: We should add a link to the release notes here with text like 'For changes in older versions, see the release notes'

File: spec.md (hunk 2 above)
Comment (mattmcneeney) [inline spec.md]: @pmorie what do you mean?

Comment (mattmcneeney) [inline spec.md]: @pmorie I see what you mean now, but I'm not going to add that to the release proposal as it is an existing issue with the spec. We can fix that in a future PR (fancy doing that one? :) )

File: spec.md (hunk 3 above)
Comment (mattmcneeney) [inline spec.md]: fixed

File: spec.md (hunk 4 above)
Comment (mattmcneeney) [inline spec.md]: done

File: release-notes.md (hunk 1 above)
Comment (mattmcneeney) [inline release-notes.md]: I wasn't intending it to be complete, but I think it should contain all changes that aren't basic formatting / typos. Updating now.

**Comment by mattmcneeney** [comment]
@pmorie updated!

File: spec.md (hunk 2 above)
Comment (shalako) [inline spec.md]: I agree with Matt.

**Comment by shalako** [comment]
LGTM

**Comment by mattmcneeney** [comment]
Calling all PMC members: @vaikas-google @fmui @zrob @avade @angarg12 @duglin

**Comment by avade** [comment]
As the release proposer and as I paired on the release notes, I will hold
off giving my official looks good.

But I think it looks good now.

**Comment by zrob** [comment]
LGTM

File: release-notes.md [hunk 5]
```diff
@@ -1,5 +1,17 @@
 # Service Broker API Release Notes #
 
+## <a id='2-12'></a>v2.12 ##
+2017-06-13
+
+* Added `context` field to request body for provisioning a service instance (`PUT /v2/service_instances/:instance_id`) and updating a service instance (`PATCH /v2/service_instances/:instance_id`). Also added the [profile.md](https://github.com/openservicebrokerapi/servicebroker/blob/master/profile.md) file describing RECOMMENDED usage patterns for environment-specific extensions and variations. `context` will replace the `organization_guid` and `space_guid` request fields in future versions of the specification. In the interim, both SHOULD be used to ensure interoperability with old and new implementations.
+* The specification now uses RFC2119 keywords to make it clearer when things are REQUIRED rather than OPTIONAL or RECOMMENDED.
+* Request and response parameters of type string, if present, MUST be a non-empty string.
+* Cleaned up text around status codes in responses, clarifying when they MUST be returned.
```
Comment (pmorie) [inline release-notes.md]: I would make this a sublist of 'Numerous clarifications' and remove the various 'clarify' variations in each bullet

File: spec.md (hunk 2 above)
Comment (pmorie) [inline spec.md]: Oh, my bad, I see that those sections omitted the header.  Yeah, I'll fix that in a later PR.

**Comment by duglin** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Comment by shalako** [comment]
Merging this PR. Let's fix any minor issues in a subsequent PR.

**Pull request merged**